internal/test/framework/framework.go

```diff
@@ -142,6 +158,79 @@ func (f *TestFramework) getKubeClient() error {
 	return nil
 }
 
+// GetNode returns a pointer to the node object associated with the external IP provided
+func (f *TestFramework) GetNode(externalIP string) (*v1.Node, error) {
+	var matchedNode *v1.Node
+
+	nodes, err := f.K8sclientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not get list of nodes")
+	}
+	if len(nodes.Items) == 0 {
+		return nil, fmt.Errorf("no nodes found")
+	}
+
+	// Find the node that has the given IP
+	for _, node := range nodes.Items {
+		for _, address := range node.Status.Addresses {
+			if address.Type == "ExternalIP" && address.Address == externalIP {
+				matchedNode = &node
+				break
+			}
+		}
+		if matchedNode != nil {
+			break
+		}
+	}
+	if matchedNode == nil {
+		return nil, fmt.Errorf("could not find node with IP: %s", externalIP)
+	}
+	return matchedNode, nil
+}
+
+// k8sVersionToOpenShiftVersion converts a Kubernetes minor version to an OpenShift version in format
+// "major.minor". This function works under the assumption that for OpenShift 4, every OpenShift minor version increase
+// corresponds with a kubernetes minor version increase
+func k8sVersionToOpenShiftVersion(k8sMinorVersion int) (string, error) {
+	openshiftMajorVersion := "4"
+	baseKubernetesMinorVersion := 16
+	baseOpenShiftMinorVersion := 3
+
+	if k8sMinorVersion < baseKubernetesMinorVersion {
+		return "", fmt.Errorf("kubernetes minor version %d is not supported", k8sMinorVersion)
+	}
+
+	// find how many minor versions past OpenShift 4.3 we are
+	versionIncrements := k8sMinorVersion - baseKubernetesMinorVersion
+	openShiftMinorVersion := strconv.Itoa(versionIncrements + baseOpenShiftMinorVersion)
+	return openshiftMajorVersion + "." + openShiftMinorVersion, nil
+}
+
+// getClusterVersion gets the OpenShift cluster version in major.minor format
```

Comment:
Please call out somewhere, why we are doing this i.e. why we are unable to use the cluster version returned by `oc version` or `oc get clusterversion version` in CI.